test/e2e/alcotest/passing/verbose_newlines.ml

```diff
@@ -0,0 +1,29 @@
+(** Reproduction for https://github.com/mirage/alcotest/issues/225, testing the
+    interaction between `--verbose` and newlines in the test stdout. *)
+
+let () =
+  Alcotest.run ~verbose:true "suite-name"
+    [
+      ( "alpha",
+        [
+          Alcotest.test_case "0 newlines" `Quick (fun () ->
+              Format.printf "Print inside alpha");
+        ] );
+      ( "beta",
+        (* Progressive result reporting is broken for this test, since the
+           carriage return happens on the wrong line. *)
+        [
+          Alcotest.test_case "1 newline" `Quick (fun () ->
+              Format.printf "Print inside beta\n");
+        ] );
+      ( "gamma",
+        [
+          (* Reporting is also broken here. Even worse, some of the test std.out
+             is clipped by the eventual result of the test. *)
+          Alcotest.test_case "1 newline + long line" `Quick (fun () ->
+              Format.printf
```

Comment:
Can you add a few `Alcotest.check` tests too?

Reply:
Added in https://github.com/mirage/alcotest/pull/226/commits/6478c96cd89e5129abbf0744d543af9a5d427aba.

This behaviour will change when https://github.com/mirage/alcotest/pull/228 is merged, since it's currently observing the `ASSERT` bug right now.